.ci/run-link-check-plugin.sh

```diff
@@ -7,8 +7,34 @@ pwd
 uname -a
 mvn --version
 curl --fail-with-body -I https://sourceforge.net/projects/checkstyle/
-mvn -e --no-transfer-progress clean site -Dcheckstyle.ant.skip=true -DskipTests -DskipITs \
-   -Dpmd.skip=true -Dspotbugs.skip=true -Djacoco.skip=true -Dcheckstyle.skip=true
+
+# Function to calculate the replacement string based on the file depth
+calculate_replacement() {
+    local file_path="$1"
+    local relative_path="${file_path#src/main/java/}"
+    local depth=$(echo "$relative_path" | grep -o "/" | wc -l)
+    local replacement=""
+    for ((i=0; i<depth; i++)); do
+        if ((i > 0)); then
+            replacement="/$replacement"
+        fi
+        replacement="..$replacement"
+    done
+    echo "../$replacement"
+}
+
+# convert all javadoc website links to relative links
+grep -rl "https://checkstyle.org" src/main/java | while read -r file; do
+    replacement=$(calculate_replacement "$file")
+    sed -i "s|https://checkstyle.org|$replacement|g" "$file"
+done
+
+mvn -e --no-transfer-progress -Pno-validations clean site -Dlinkcheck.skip=false \
+   -Dmaven.javadoc.skip=false
```

Comment:
This command was replaced using `no-validations` so we don't have to manually disable checks, especially if one is added, and we can clearly see what is being enabled.

Reply:
I think we can make sepatate PR for this update only and merge it quickly.

Reply:
Yeah, this is good